.github/workflows/unit-tests.yml

```diff
@@ -305,6 +305,45 @@ jobs:
       group: ${{ github.event_name == 'push' && github.run_number || github.ref }}-musl
       cancel-in-progress: true
 
+  ASAN_UBSAN:
+    runs-on: ubuntu-22.04
+    timeout-minutes: 90
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Set up Conda
+      uses: ./.github/actions/setup-conda
+      with:
+        environment-file: ci/deps/actions-311-numpydev.yaml
+
+    - name: Build Pandas
+      id: build
+      uses: ./.github/actions/build_pandas
+      with:
+        meson_args: --config-settings=setup-args="-Db_sanitize=address,undefined"
+        cflags_adds: -fno-sanitize-recover=all
```

Comment:
Without this switch ASAN will in some cases recover and continue, which doesn't help CI to actually error out when issues occur (at least not with pytest-xdist)

Reply:
If we get more predictable/consistent results without xdist, I think we should not run the tests in parallel